pennylane/transforms/sign_expand.py

```diff
@@ -0,0 +1,327 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the sign (and xi) decomposition tape transform, implementation of ideas from arXiv:2207.09479
+"""
+# pylint: disable=protected-access
+import json
+from os import path
+import pennylane as qml
+from pennylane import numpy as np
+
+# TODO: This part up here (ControlledPauliEvolution and MultiCRZ) probably should live somewhere else, not sure if properly implementing these as gates inside pennylane is worthwhile
+
+
+def ControlledPauliEvolution(theta, wires, pauli_word, ancillas):
```

Comment:
How about calling this `CPauliRot`.  That way we are using consistent terminology and naming schemes. Or is this not a Controlled `PauliRot`?

I also have a preference for using the term "auxiliary" or "aux" instead of "ancilla", as "ancilla" means "female slave".

Are the "ancillas" control qubits in this case anyway?

Reply:
Ancillas renamed  to `controls`

I think CPauliRot is a bit misleading as one assumes this is just a controlled version of PauliRot (which it is not, this is
e−iZ⊗(O)t where O is the generator of the pauliword)